src/Uno.UI/UI/Xaml/Controls/Image/Image.cs

```diff
@@ -56,13 +56,20 @@ public static new class TraceProvider
 			public const int Image_SetImageStop = 6;
 		}
 
-		protected virtual void OnImageFailed(ImageSource imageSource)
+		private protected void OnImageFailed(ImageSource imageSource, Exception exception)
 		{
 			if (this.Log().IsEnabled(Uno.Foundation.Logging.LogLevel.Debug))
 			{
 				this.Log().Debug(this.ToString() + " Image failed to open");
```

Comment:
more context is always better:
```suggestion
				this.Log().Debug($"Image '{this}' failed to open: {exception}");
```